dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/JaegerSecureOtlpTest.java

```diff
@@ -0,0 +1,135 @@
+import io.openliberty.microprofile.telemetry.internal.utils.jaeger.JaegerContainer;
+import io.openliberty.microprofile.telemetry.internal.utils.jaeger.JaegerQueryClient;
+
+/**
+ * Test exporting traces to a Jaeger server with OTLP
+ */
+@RunWith(FATRunner.class)
+public class JaegerSecureOtlpTest extends JaegerBaseTest {
+
+    @ClassRule
+    public static JaegerContainer jaegerContainer = new JaegerContainer(getCertificate(), getKey()).withLogConsumer(new SimpleLogConsumer(JaegerBaseTest.class, "jaeger"));
+
+    public static JaegerQueryClient client;
+
+    private static String key;
+    private static String cert;
+
+    private static boolean createdSSLStuff = false;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+
+        client = new JaegerQueryClient(jaegerContainer);
+
+        server.addEnvVar(TestConstants.ENV_OTEL_TRACES_EXPORTER, "otlp");
+        server.addEnvVar(TestConstants.ENV_OTEL_EXPORTER_OTLP_ENDPOINT, jaegerContainer.getSecureOtlpGrpcUrl());
+
+        server.addEnvVar(TestConstants.ENV_OTEL_SERVICE_NAME, "Test service");
+        server.addEnvVar(TestConstants.ENV_OTEL_BSP_SCHEDULE_DELAY, "100"); // Wait no more than 100ms to send traces to the server
+        server.addEnvVar(TestConstants.ENV_OTEL_SDK_DISABLED, "false"); //Enable tracing
+
+        //enable TLS
+        //server.addEnvVar(TestConstants.ENV_COLLECTOR_OTLP_GRPC_TLS_ENABLED, "true"); //Don't think is required
+
+        server.addEnvVar(TestConstants.ENV_OTEL_EXPORTER_OTLP_CERTIFICATE, cert);
+
+        //server.addEnvVar(TestConstants.ENV_OTEL_EXPORTER_OTLP_CLIENT_KEY, key.getAbsolutePath());
+        //server.addEnvVar(TestConstants.ENV_OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE, CACert.getAbsolutePath());
+
+        // Construct the test application
+        WebArchive jaegerTest = ShrinkWrap.create(WebArchive.class, "spanTest.war")
+                                          .addClass(TestResource.class);
+        ShrinkHelper.exportAppToServer(server, jaegerTest, SERVER_ONLY);
+        server.startServer();
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        server.stopServer();
+    }
+
+    @AfterClass
+    public static void closeClient() throws Exception {
+        client.close();
+    }
+
+    @Override
+    protected JaegerQueryClient getJaegerClient() {
+        return client;
+    }
+
+    private static String getKey() {
+        generateSSLStuff();
+        return key;
+    }
+
+    private static String getCertificate() {
+        generateSSLStuff();
+        return cert;
+    }
+
+    private synchronized static void generateSSLStuff() {
+        if (createdSSLStuff) {
+            return;
+        }
+
+        try {
+            KeyPair generatedKeyPair = SSLUtils.generateKeyPair();
+
+            String dockerIP = DockerClientFactory.instance().dockerHostIpAddress();
+            String dnName = "O=Evil Inc, CN="+dockerIP+", L=Toronto,C=CA";
```

Comment:
In case the certificate escapes and ends up somewhere odd, the dnName should probably indicate that it's a test certificate.

Reply:
Evil Inc not enough? 

I'll change it to Evil Inc's Test Certificate.